File: desktopclient/src/main/scala/com/goexp/galgame/gui/task/game/panel/group/node/DateItem.java
package com.goexp.galgame.gui.task.game.panel.group.node;

import com.goexp.galgame.common.model.DateRange;

import java.time.LocalDate;

public class DateItem extends DefaultItem {

    public final DateRange range;
    public final DateType dateType;

    public DateItem(String title, LocalDate start, LocalDate end, int count, DateType dateType) {
        this(title, new DateRange(start, end), count, dateType);
    }


    public DateItem(String title, DateRange range, int count, DateType dateType) {
        super(title, count);
        this.range = range;
        this.dateType = dateType;
    }

    public enum DateType {
        YEAR,
        MONTH
    }
}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/task/game/panel/group/node/DateItem.scala
package com.goexp.galgame.gui.task.game.panel.group.node

import com.goexp.galgame.common.model.DateRange

import java.time.LocalDate
import scala.beans.BeanProperty

case class DateItem(@BeanProperty title: String, range: DateRange, @BeanProperty count: Int, @BeanProperty dateType: DateType) extends DataItem {

  def this(title: String, start: LocalDate, end: LocalDate, count: Int, dateType: DateType) = {
    this(title, DateRange(start, end), count, dateType)
  }
}